File: SoccerPlayerApi/SoccerPlayerApi/Dtos/Aggregations/AggregationDto.cs
namespace SoccerPlayerApi.Dtos.Aggregations;

public class AggregationDto
{
    public int Id { get; set; }
    public string Label { get; set; } = string.Empty;

    public int LevelId { get; set; }
    public string LevelLabel { get; set; } = string.Empty;

    public int DimensionId { get; set; }
    public string DimensionLabel { get; set; } = string.Empty;

    public int? MotherAggregationId { get; set; }
    public string? MotherAggregationValue { get; set; }

    public int? MotherLevelId { get; set; }
    public string? MotherLevelLabel { get; set; }
}

File: SoccerPlayerApi/clientapp/src/app/models/aggregations/aggregationDto.ts
export class AggregationDto {
    id: number;
    label: string = '';

    levelId: number;
    levelLabel: string = '';

    dimensionId: number;
    dimensionLabel: string = '';

    motherAggregationId?: number;
    motherAggregationValue?: string;

    motherLevelId?: number;
    motherLevelLabel?: string;

    constructor(
        id: number,
        label: string = '',
        levelId: number,
        levelLabel: string = '',
        dimensionId: number,
        dimensionLabel: string = '',
        motherAggregationId?: number,
        motherAggregationValue?: string,
        motherLevelId?: number,
        motherLevelLabel?: string
    ) {
        this.id = id;
        this.label = label;
        this.levelId = levelId;
        this.levelLabel = levelLabel;
        this.dimensionId = dimensionId;
        this.dimensionLabel = dimensionLabel;
        this.motherAggregationId = motherAggregationId;
        this.motherAggregationValue = motherAggregationValue;
        this.motherLevelId = motherLevelId;
        this.motherLevelLabel = motherLevelLabel;
    }
}
